src/sentry/eventstream/snuba.py

```diff
@@ -1,15 +1,251 @@
 from __future__ import absolute_import
 
+import logging
+from datetime import datetime
+from uuid import uuid4
+
+import pytz
 import urllib3
 
-from sentry.eventstream.kafka import KafkaEventStream
-from sentry.eventstream.kafka.backend import EVENT_PROTOCOL_VERSION
+from sentry import quotas
+from sentry.eventstream.base import EventStream
 from sentry.utils import snuba, json
+from sentry.utils.safe import get_path
+
+
+logger = logging.getLogger(__name__)
+
+
+# Version 1 format: (1, TYPE, [...REST...])
+#   Insert: (1, 'insert', {
+#       ...event json...
+#   }, {
+#       ...state for post-processing...
+#   })
+#
+#   Mutations that *should be ignored*: (1, ('delete_groups'|'unmerge'|'merge'), {...})
+#
+#   In short, for protocol version 1 only messages starting with (1, 'insert', ...)
+#   should be processed.
+
+# Version 2 format: (2, TYPE, [...REST...])
+#   Insert: (2, 'insert', {
+#       ...event json...
+#   }, {
+#       ...state for post-processing...
+#   })
+#   Delete Groups: (2, '(start_delete_groups|end_delete_groups)', {
+#       'transaction_id': uuid,
+#       'project_id': id,
+#       'group_ids': [id2, id2, id3],
+#       'datetime': timestamp,
+#   })
+#   Merge: (2, '(start_merge|end_merge)', {
+#       'transaction_id': uuid,
+#       'project_id': id,
+#       'previous_group_ids': [id2, id2],
+#       'new_group_id': id,
+#       'datetime': timestamp,
+#   })
+#   Unmerge: (2, '(start_unmerge|end_unmerge)', {
+#       'transaction_id': uuid,
+#       'project_id': id,
+#       'previous_group_id': id,
+#       'new_group_id': id,
+#       'hashes': [hash2, hash2]
+#       'datetime': timestamp,
+#   })
+#   Delete Tag: (2, '(start_delete_tag|end_delete_tag)', {
+#       'transaction_id': uuid,
+#       'project_id': id,
+#       'tag': 'foo',
+#       'datetime': timestamp,
+#   })
+
+
+class SnubaProtocolEventStream(EventStream):
+
+    # Beware! Changing this protocol (introducing a new version, or the message
+    # format/fields themselves) requires consideration of all downstream
+    # consumers. This includes the post-processing relay code!
+    EVENT_PROTOCOL_VERSION = 2
```

Comment:
This admonition is now detached from the actual protocol definition … seems wrong?